### PR TITLE
Bail out early when there is no property

### PR DESCRIPTION
**Pull request by rwlbuis**
Bail out early when there is no property for the event handler, since
otherwise we get timeouts later on.

### CODE REVIEW

File: mathml/relations/html5-tree/math-global-event-handlers.tentative.html [hunk 1]
```diff
@@ -28,9 +28,9 @@
 
       for (const name of names) {
         const withoutOn = name.substring(2);
+        const location = MathMLElement.prototype;
```
Comment (fred-wang) [inline mathml/relations/html5-tree/math-global-event-handlers.tentative.html]: location is used elsewhere with different types, not sure that's a very good name. Also you probably don't need to factor out this variable.

File: mathml/relations/html5-tree/math-global-event-handlers.tentative.html [hunk 2]
```diff
@@ -42,6 +42,9 @@
           );
         }, `${name}: must be on the appropriate locations for GlobalEventHandlers`);
 
+        if (!location.hasOwnProperty(name))
```
Comment (fred-wang) [inline mathml/relations/html5-tree/math-global-event-handlers.tentative.html]: I think this is effectively skipping tests (and hence reducing the test count)?
Would it be possible instead to modify the async_test to add a t.step() call that asserts MathMLElement.prototype.hasOwnProperty(name) ? IIUC, failure would just stop the test without waiting for the t.step_func_done().

BTW, I think we want the same for clipboard-event-handlers.tentative.html

**Review by fred-wang — CHANGES_REQUESTED** [review]

File: mathml/relations/html5-tree/math-global-event-handlers.tentative.html (hunk 2 above)
Comment (rwlbuis) [reply mathml/relations/html5-tree/math-global-event-handlers.tentative.html]: Funnily enough this exposes another test and increases test count:
-Found 244 tests; 238 PASS, 6 FAIL, 0 TIMEOUT, 0 NOTRUN.
+Found 297 tests; 295 PASS, 2 FAIL, 0 TIMEOUT, 0 NOTRUN.

I don't see the point in running subtests we have already deduced will fail.

Comment (fred-wang) [reply mathml/relations/html5-tree/math-global-event-handlers.tentative.html]: Mmh, it seems you are comparing against your initial commit. This is what I get comparing with ToT:

Current ToT
Found 305 tests
295 Pass
8 Fail
2 Timeout

With the patch below
Found 305 tests
295 Pass
10 Fail

diff --git a/mathml/relations/html5-tree/math-global-event-handlers.tentative.html b/mathml/relations/html5-tree/math-global-event-handlers.tentative.html
index e96feeaf09..859feff422 100644
--- a/mathml/relations/html5-tree/math-global-event-handlers.tentative.html
+++ b/mathml/relations/html5-tree/math-global-event-handlers.tentative.html
@@ -127,2 +127,5 @@
         async_test(t => {
+          t.step(function() {
+              assert_true(MathMLElement.prototype.hasOwnProperty(name));
+          });
           const element = document.createElementNS(

I think we should not reduce the number of tests (305 tests here) and moreover we should keep the same number of tests for all browsers. Skipping tests according to whether the property is supported by the browser will make the percentage of pass incomparable in the implementation report. I we cannot avoid this issue when the browser crashes but here we have a simple solution.

**Comment by fred-wang** [comment]
@rwlbuis Can you please do the same for mathml/relations/html5-tree/clipboard-event-handlers.tentative.html to avoid timeouts?

**Review by fred-wang — APPROVED** [review]
LGTM, thanks!

**Comment by bzbarsky** [comment]
@rwlbuis do you happen to know why this is an async_test to start with?  Event dispatch is sync, so it seems like this could just be a sync test that fires the event and then checks that the listener was called...

**Comment by fred-wang** [comment]
cc @bkardell

**Comment by fred-wang** [comment]
I opened https://github.com/web-platform-tests/wpt/issues/19500

**Pull request merged**